src/actions/modifiers.ts

```diff
@@ -1,22 +1,25 @@
 import { ActionCreator } from 'redux'
-import { AnyAction, ThunkAction } from 'src/store'
+import { ThunkAction } from 'src/store'
 import { AppError } from '../lib/errors'
+import { toggleLoading } from './account'
 
 /**
- * Curried function that wraps a {@link ThunkAction} with two calls to the provided loadingAction
- * {@link ActionCreator} (called with "true" before the wrapped action is dispatched, and with "false" after)
- * @param loadingAction - An action creator to create the actions to be dispatched before and after the wrappedAction-
+ * Action modifier that wraps a {@link ThunkAction} with calls to {@link toggleLoading},
+ * to show the AppLoading spinner while the action is being executed
  * @param wrappedAction - The thunkAction to be wrapped
- * @example dispatch(withLoading(toggleLoading)(saveClaims))
+ * @example dispatch(withLoading(saveClaims))
  */
-export const withLoading = (loadingAction: ActionCreator<AnyAction>) => (
+export const withLoading = (
   wrappedAction: ThunkAction,
 ): ThunkAction => async dispatch => {
   try {
-    dispatch(loadingAction(true))
+    dispatch(toggleLoading(true))
     return await dispatch(wrappedAction)
   } finally {
-    dispatch(loadingAction(false))
+    // NOTE: timeout is a hack to avoid flashing when navigating, as the new
+    // screen will have not been loaded yet but the AppLoading will be taken
+    // down, which will flash the old screen shortly
+    setTimeout(() => dispatch(toggleLoading(false)), 100)
```

Comment:
Where is this useful? I tried without the timeout and I don't see any differences.

Reply:
Also, we can keep the previous version of `withLoading` and export a function, e.g. `withLoadingGeneric` or something like that defined as
`withLoadingGeneric = withLoading(toggleLoading)`

Reply:
This avoid flashing of the previous screen, as I mention in the comment.
One specific example is after reading a QR code, before going to the Consent screen, the QR screen will flash for a split second. Note that this is very device dependent, since if the change is quick enough you will not really see that flash.

Reply:
As for `withLoading` yes I agree, will change that